html/embed/index.js

```diff
@@ -139,4 +171,96 @@ function handleTsParam(ts) {
   }
 
   return ts;
-}
\ No newline at end of file
+}
+
+class TimeoutError extends Error {
+  constructor(message) {
+    super(message);
+    this.name = "TimeoutError";
+    this.timeOutError = true;
+  }
+}
+
+/**
+ * Waits for a given element to be in the DOM and returns it.
+ * Wait is based on `requestAnimationFrame`: timeout is approximately 60 seconds (60 x 60 frames per seconds).
+ *
+ * Takes a function querying the DOM for a single element as an argument
+ *
+ * @param {function} selectorFunction - Function to be run to find the element.
+ * @returns {Promise<HTMLElement>} Reference to the element that was found.
+ */
+async function waitForElement(selectorFunction) {
+  const maxPauseSeconds = 60;
+  let tries = maxPauseSeconds * 60;  // we expect a repaint rate of ~60 times a second, per https://developer.mozilla.org/en-US/docs/Web/API/window/requestAnimationFrame
+  let elem = null;
+
+  while (!elem && tries > 0) {
+    // Sleep efficiently until the next repaint
+    const pause = await new Promise(resolve => requestAnimationFrame(resolve));
+    cancelAnimationFrame(pause);
+
+    // Look for the target element
+    try {
+      elem = selectorFunction();
+    }
+    catch (err) {
+      if (!err.message.includes('null')) {
+        throw err;
+      }
+    }
+    if (!elem) {
+      tries -= 1;
+    }
+  }
+
+  if (elem) {
+    return elem;
+  }
+
+  throw new TimeoutError(`Did not find element in ${maxPauseSeconds}s`);
+}
+
+/**
+ * Async helper function for handling `overrideElementAttribute` messages.
+ * Posts `overrideElementAttribute` back to the parent frame on failure.
+ *
+ * @param {string} origin - The origin of the posted message (see: https://developer.mozilla.org/en-US/docs/Web/API/MessageEvent/origin).
+ * @param {HTMLElement} player - The <replay-web-page> element.
+ * @param {HTMLElement} parent - https://developer.mozilla.org/en-US/docs/Web/API/Window/parent
```

Comment:
Oh I missed this in my earlier suggestion, I think this line should be:
```suggestion
 * @param {Window} parent - https://developer.mozilla.org/en-US/docs/Web/API/Window/parent
```